engine/schema/src/org/apache/cloudstack/acl/RolePermissionVO.java

```diff
@@ -0,0 +1,109 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.cloudstack.acl;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.util.UUID;
+
+@Entity
+@Table(name = "role_permissions")
+public class RolePermissionVO implements RolePermission {
```

Comment:
@koushik-das with the current static commands.properties based approach, new APIs are enabled for role by developer by making changes in commands.properties file. During installation it is seen that any pre-existing commands.properties file does not get overwritten, so admin need to enable/change API for roles manually in commands.properties file.

With this feature, the API developer when adding new API will enable the API for the default role type using the authorized field in \@APICommand annotation. This is strictly to enforce default behaviour when there are no allow or deny rule for that API. The admin when upgrading to a new version to get the new APIs etc won't have to manually allow/deny them if authorized annotation is present. In general, the release notes should have list of new APIs that admins read and decide if they want to update/add/modify role permissions.

I'm not sure what you mean by `Role refers RolePermissions`, as of now each role has a list of role permissions linked to it if that's what you're asking.

This has been documented in the FS under the section E: https://cwiki.apache.org/confluence/display/CLOUDSTACK/Dynamic+Role+Based+API+Access+Checker+for+CloudStack